flow/project.py

```diff
@@ -277,36 +277,49 @@ class _AggregatesCursor:
 
     """
 
-    # This class currently only handles aggregates of size 1 (single jobs).
-
     def __init__(self, project, filter=None, doc_filter=None):
         self._project = project
         self._filter = filter
         self._doc_filter = doc_filter
-        self._jobs_cursor = JobsCursor(project, filter, doc_filter)
+        # If no filter or doc_filter is provided by the user, then select every
+        # aggregate present in the flow project. If filter or doc_filter is provided,
+        # then use the jobs returned via JobsCursor instance.
+        if filter is None and doc_filter is None:
+            self._cursor = project._group_to_aggregate_store.inverse.keys()
+        else:
+            self._cursor = JobsCursor(project, filter, doc_filter)
 
     def __eq__(self, other):
-        return self._jobs_cursor == other._jobs_cursor
+        if not (
+            isinstance(other, type(self))
+            and isinstance(other._cursor, type(self._cursor))
+        ):
+            return NotImplemented
+        return self._cursor == other._cursor
 
     def __contains__(self, aggregate):
-        if len(aggregate) != 1:
-            # Exit early if this is not an aggregate of 1 job
-            return False
         if self._filter is None and self._doc_filter is None:
-            # Using the Project's __contains__ method is fastest if no
-            # filtering is needed. This is a backport of PR 449 to signac that
-            # optimizes the JobsCursor __contains__ method, and is required for
-            # sufficient performance with earlier versions of signac.
-            return aggregate[0] in self._project
-        # Slow path: requires O(N) iteration over the JobsCursor
-        return aggregate[0] in self._jobs_cursor
+            return any(
+                get_aggregate_id(aggregate) in aggregate_store
+                for aggregate_store in self._cursor
+            )
+        else:
+            return aggregate[0] in self._cursor
```

Comment:
@vyasr has an idea for improvement.

Reply:
Here's my suggestion. Currently most methods of `AggregatesCursor` behave differently based on the exact same criterion, namely whether or not there are filters, but semantically the presence of a filter indicates something more because if we have a filter we only iterate over a `JobsCursor`, except that we wrap the returned jobs in a tuple. That to me indicates that we should have (very small) subclasses to distinguish the two cases, with one case being a true `AggregatesCursor`, whereas the other is very thin wrapper around a `JobsCursor` that returns tuples instead.